client.lua

```diff
@@ -89,11 +91,64 @@ local function hasHarness(items)
     harness = _harness
 end
 
+-- Weapon Stress [Haroki]
+local function IsWhitelistedWeaponStress(weapon)
+    if weapon then
+        for _, v in pairs(config.WhitelistedWeaponStress) do
+            if weapon == v then
+                return true
+            end
+        end
+    end
+    return false
+end
+
+local hasWeapon = false
+
+local function startWeaponStressThread(weapon)
+    if IsWhitelistedWeaponStress(weapon) then hasWeapon = false; return end
+
+    CreateThread(function()
+        while hasWeapon do
+            local ped = cache.ped
+            if IsPedShooting(ped) then
+                if math.random() < config.StressChance then
+                    TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
+                end
+            end
+            Wait(0)
+        end
+    end)
+end
+
+AddEventHandler('ox_inventory:currentWeapon', function(currentWeapon)
+    if not currentWeapon then
+        hasWeapon = false
+        return
+    end
+
+    local curGun = currentWeapon.name
+    hasWeapon = true
+    startWeaponStressThread(curGun)
+end)
+
+-- Weapon Stress end
+AddEventHandler('onResourceStart', function (resource)
+    if not resource == GetCurrentResourceName() or not LocalPlayer.state.isLoggedIn then return end
+    local Weapon = OxInventory:getCurrentWeapon()
+    PlayerData = QBCore.Functions.GetPlayerData()
+    PlayerLoaded = true
+    if Weapon then startWeaponStressThread(Weapon.name) end
+end)
```

Comment:
We use onResourceStart to ensure the resource has started and the server has started as well before executing any code, otherwise the event wouldn't have any use and wouldn't exist

Reply:
The point of this code is to be executed when the resource restarts. This runs on the client meaning that the server has already started. When the code runs that also means that the resource has started. No other checks for other resources have to be done since we check whether the client has logged in.

The `onResourceStart` has other use cases (it doesn't only trigger for the current resource, after all), and I don't think this is one of them.

Reply:
What I'm afraid of, is this code executing before the client side of ox_inventory has started

Reply:
But wouldn't `LocalPlayer.state.isLoggedIn` be false?

Reply:
You're absolutely right, I havem't thought that through lmao